stmhal/spi.c

```diff
@@ -747,18 +784,22 @@ STATIC const mp_map_elem_t pyb_spi_locals_dict_table[] = {
     /// \constant SLAVE - for initialising the bus to slave mode
     /// \constant MSB - set the first bit to MSB
     /// \constant LSB - set the first bit to LSB
+    /// \constant 2WIRE - 2 wire full duplex mode
+    /// \constant 1WIRE_RX - 1 wire unidirectional receive
+    /// \constant 1WIRE_BI - 1 wire bidirectional mode
+    /// \constant CS_SOFT - slave select via sw
+    /// \constant CS_HARD_INPUT - slave select via hw (used by master devices only)
+    /// \constant CS_HARD_OUTPUT - slave select via hw (used by slaves for multimaster cap.)
     { MP_OBJ_NEW_QSTR(MP_QSTR_MASTER), MP_OBJ_NEW_SMALL_INT(SPI_MODE_MASTER) },
     { MP_OBJ_NEW_QSTR(MP_QSTR_SLAVE),  MP_OBJ_NEW_SMALL_INT(SPI_MODE_SLAVE) },
     { MP_OBJ_NEW_QSTR(MP_QSTR_MSB),    MP_OBJ_NEW_SMALL_INT(SPI_FIRSTBIT_MSB) },
     { MP_OBJ_NEW_QSTR(MP_QSTR_LSB),    MP_OBJ_NEW_SMALL_INT(SPI_FIRSTBIT_LSB) },
-    /* TODO
-    { MP_OBJ_NEW_QSTR(MP_QSTR_DIRECTION_2LINES             ((uint32_t)0x00000000)
-    { MP_OBJ_NEW_QSTR(MP_QSTR_DIRECTION_2LINES_RXONLY      SPI_CR1_RXONLY
-    { MP_OBJ_NEW_QSTR(MP_QSTR_DIRECTION_1LINE              SPI_CR1_BIDIMODE
-    { MP_OBJ_NEW_QSTR(MP_QSTR_NSS_SOFT                    SPI_CR1_SSM
-    { MP_OBJ_NEW_QSTR(MP_QSTR_NSS_HARD_INPUT              ((uint32_t)0x00000000)
-    { MP_OBJ_NEW_QSTR(MP_QSTR_NSS_HARD_OUTPUT             ((uint32_t)0x00040000)
-    */
+    { MP_OBJ_NEW_QSTR(MP_QSTR_TWO_WIRE), MP_OBJ_NEW_SMALL_INT(SPI_DIRECTION_2LINES) },
+    { MP_OBJ_NEW_QSTR(MP_QSTR_ONE_WIRE_RX), MP_OBJ_NEW_SMALL_INT(SPI_CR1_RXONLY) },
+    { MP_OBJ_NEW_QSTR(MP_QSTR_ONE_WIRE_BI), MP_OBJ_NEW_SMALL_INT(SPI_CR1_BIDIMODE) },
```

Comment:
These two macors are handled in HAL. Better introduce SPI_DIRECTION_1LINE. 
For a working implementation see #2058.